torch/distributions/transforms.py

```diff
@@ -0,0 +1,242 @@
+        """
+        raise NotImplementedError
+
+
+class InverseTransform(Transform):
+    """
+    Inverts a single :class:`Transform`.
+    """
+    def __init__(self, bijector):
+        self.bijector = bijector
+
+    @constraints.dependent_property
+    def domain(self):
+        return self.bijector.codomain
+
+    @constraints.dependent_property
+    def codomain(self):
+        return self.bijector.domain
+
+    def forward(self, x):
+        return self.bijector.inverse(x)
+
+    def inverse(self, y):
+        return self.bijector.forward(y)
+
+    def log_abs_det_jacobian(self, x, y):
+        return -self.bijector.log_abs_det_jacobian(y, x)
+
+
+class ExpTransform(Transform):
+    """
+    Transform via the mapping `y = exp(x)`.
+    """
+    domain = constraints.real
+    codomain = constraints.positive
+
+    def _forward(self, x):
+        return x.exp()
+
+    def _inverse(self, y):
+        return y.log()
+
+    def log_abs_det_jacobian(self, x, y):
+        return x
+
+
+class SigmoidTransform(Transform):
+    """
+    Transform via the mapping `y = sigmoid(x)` and `x = logit(y)`.
+    """
+    domain = constraints.real
+    codomain = constraints.unit_interval
+
+    def _forward(self, x):
+        return sigmoid(x)
+
+    def _inverse(self, y):
+        return y.log() - (-y).log1p()
+
+    def log_abs_det_jacobian(self, x, y):
+        return -(y.reciprocal() + (1 - y).reciprocal()).log()
+
+
+class AbsTransform(Transform):
+    """
+    Transform via the mapping `y = abs(x)`
+    """
+    domain = constraints.real
+    codomain = constraints.positive
+
+    def _forward(self, x):
+        return x.abs()
+
+
+class AffineTransform(Transform):
+    """
+    Transform via the pointwise affine mapping `y = loc + scale * x`.
+
+    Args:
+        loc (Tensor or Variable): Location parameter.
+        scale (Tensor or Variable): Scale parameter.
+        event_dim (int): Optional size of `event_shape`. This should be zero
+            for univariate random variables, 1 for distributions over vectors,
+            2 for distributions over matrices, etc.
+    """
+    domain = constraints.real
+    codomain = constraints.real
+
+    def __init__(self, loc, scale, event_dim=0):
+        super(AffineTransform, self).__init__()
+        self.loc = loc
+        self.scale = scale
+        self.event_dim = event_dim
+
+    def __eq__(self, other):
+        return (type(other) is AffineTransform) and self.loc.eq(other.loc).all() and self.scale.eq(other.scale).all()
+
+    def _forward(self, x):
+        return self.loc + self.scale * x
+
+    def _inverse(self, y):
+        return (y - self.loc) / self.scale
+
+    def log_abs_det_jacobian(self, x, y):
+        result = torch.abs(self.scale).log()
+        shape = x.shape
+        if self.event_dim:
+            # NOTE: no need for contiguous here
+            result = result.view(*result.size()[:-self.event_dim], -1).sum(-1)
+            shape = shape[:-self.event_dim]

```

Comment:
To fix Python 2 compatibility you can do that:
```
result_size = result.size()[:-self.event_dim] + (-1,)
result = result.view(result_size).sum(-1)
```
Also, please remove the note (it was just a general observation, but doesn't need to be in the code)